File: MiraiCL.Core/Minecraft/Instance/Clients/Minecraft/Source.cs
using MiraiCL.Core.App.Setup;

namespace MiraiCL.Core.Minecraft.Instance.Clients.Minecraft;

public static class DownloadSourcePolicy{
    private static readonly Source Mojang = new(){
        VersionIndexV2 = "https://piston-meta.mojang.com/mc/game/version_manifest_v2.json",
        VersionIndex = "https://piston-meta.mojang.com/mc/game/version_manifest.json",
        MavenServer = "https://libraries.minecraft.net",
        JavaIndexUrl = "https://piston-meta.mojang.com/v1/products/java-runtime/2ec0cc96c44e5a76b9c8b7c39df7210883d12871/all.json",
        AssetsServer = "https://resources.download.minecraft.net",
        SourceName = "Mojang"
    };
    ///<summary>
    /// 获取版本列表下载地址（现阶段暂时只返回 Mojang 官方源的地址）
    ///</summary>
    public static Source[] GetPerferVersionSource(){
        return [Mojang];
    }
}

public record Source{
    public string? VersionIndexV2;
    public string? VersionIndex;
    public string? MavenServer;
    public string? JavaIndexUrl;
    public string? AssetsServer;
    public required string SourceName;
}

File: miraicl/core/minecraft/client/source.py
from typing import Optional

class McVersionSource:
    version_index:str
    version_index_v2:Optional[str]
    maven_server:Optional[str]
    resource_server:Optional[str]
    java_index_url:Optional[str]
    description:str

mojang = McVersionSource()

mojang.version_index = "https://piston-meta.mojang.com/mc/game/version_manifest.json"

mojang.version_index_v2 = "https://piston-meta.mojang.com/mc/game/version_manifest_v2.json"

mojang.maven_server = "https://libraries.minecraft.net"

mojang.resource_server = "https://resources.download.minecraft.net"

mojang.java_index_url = "https://piston-meta.mojang.com/v1/products/java-runtime/2ec0cc96c44e5a76b9c8b7c39df7210883d12871/all.json"

mojang.description = "Official download source."

source = {
    "Official": mojang
}